reconfiguration/src/reconfiguration_handler.cpp

```diff
@@ -199,6 +200,39 @@ bool ReconfigurationHandler::handle(const concord::messages::InstallCommand& cmd
   return true;
 }
 
+bool ReconfigurationHandler::handle(const concord::messages::GetDbCheckpointInfoRequest& req,
+                                    uint64_t,
+                                    uint32_t,
+                                    const std::optional<bftEngine::Timestamp>&,
+                                    concord::messages::ReconfigurationResponse& rres) {
+  concord::messages::GetDbCheckpointInfoStatusResponse response;
+  std::map<CheckpointId, DbCheckpointMetadata::DbCheckPointDescriptor> dbCheckpointList =
+      DbCheckpointManager::instance().getListOfDbCheckpoints();
+  for (auto& kv : dbCheckpointList) {
+    concord::messages::DbCheckpointInfo dbcpinfo_msg;
+    dbcpinfo_msg.seq_num = kv.second.lastDbCheckpointSeqNum_;
+    dbcpinfo_msg.block_id = kv.second.lastBlockId_;
+    dbcpinfo_msg.timestamp = kv.second.creationTimeSinceEpoch_.count();
+
+    response.db_checkpoint_info.push_back(dbcpinfo_msg);
+    LOG_INFO(getLogger(),
+             "GetDbCheckpointInfoStatus checkpoint id's are "
+                 << KVLOG(kv.second.lastBlockId_, kv.second.lastDbCheckpointSeqNum_));
+  }
+  rres.response = std::move(response);
+  return true;
+}
+
+bool ReconfigurationHandler::handle(const concord::messages::CreateDbCheckpointCommand& cmd,
+                                    uint64_t sequence_number,
+                                    uint32_t,
+                                    const std::optional<bftEngine::Timestamp>&,
+                                    concord::messages::ReconfigurationResponse& rres) {
+  LOG_INFO(getLogger(), "CreateDbCheckpointCommand to create dbcheckpoint at seq_num " << sequence_number);
+  DbCheckpointManager::instance().createDbCheckpoint(sequence_number);
```

Comment:
This will a blocking call, we should do an async create of db checkpoint. I think its better to add an api to create db checkpoint asynchronously.

Reply:
fixed